packages/block-library/src/button/edit.js

```diff
@@ -81,19 +83,32 @@ function ButtonEdit( props ) {
 	const { textAlign, linkTarget, placeholder, rel, style, text, url, width } =
 		attributes;
 
-	function onToggleOpenInNewTab( value ) {
-		const newLinkTarget = value ? '_blank' : undefined;
+	function updateLinkAttributes( opensInNewWindow, nofollow ) {
+		let newLinkTarget;
+		let updatedRel = rel || '';
 
-		let updatedRel = rel;
-		if ( newLinkTarget && ! rel ) {
-			updatedRel = NEW_TAB_REL;
-		} else if ( ! newLinkTarget && rel === NEW_TAB_REL ) {
-			updatedRel = undefined;
+		if ( opensInNewWindow ) {
+			newLinkTarget = NEW_TAB_TARGET;
+			updatedRel = updatedRel?.includes( NEW_TAB_REL )
+				? updatedRel
+				: updatedRel + ` ${ NEW_TAB_REL }`;
+		} else {
+			const relRegex = new RegExp( `\\b${ NEW_TAB_REL }\\s*`, 'g' );
+			updatedRel = updatedRel?.replace( relRegex, '' ).trim();
+		}
+
+		if ( nofollow ) {
+			updatedRel = updatedRel?.includes( NOFOLLOW_REL )
+				? updatedRel
+				: updatedRel + ` ${ NOFOLLOW_REL }`;
+		} else {
+			const relRegex = new RegExp( `\\b${ NOFOLLOW_REL }\\s*`, 'g' );
+			updatedRel = updatedRel?.replace( relRegex, '' ).trim();
 		}
 
 		setAttributes( {
 			linkTarget: newLinkTarget,
-			rel: updatedRel,
+			rel: updatedRel || undefined,
```

Comment:
Why `undefined` here?

Reply:
As explain [here](https://github.com/WordPress/gutenberg/pull/54110#discussion_r1319295977), since I used string concatenation, this line [let updatedRel = rel || '';](https://github.com/WordPress/gutenberg/blob/8ae150433d58ffbb0536f4bc1435b043b905b0a0/packages/block-library/src/button/edit.js#L88)  to avoid result like `undefined nofollow`

And in case `updatedRel` value is empty in the end, converting to `undefined` will ensure the HTML attribute `rel=''` is not inserted in the block HTML, am I too inflexible on this, or should we just update e2e testing snapshot to match the case?

Reply:
Ah I see yes. Thanks for that explanation 👍